### PR TITLE
[simd]: Implement v128_replace_lane instructions

### PR DESCRIPTION
**Pull request by haoyu-zc**

### CODE REVIEW

File: src/engine/x86-64/X86_64Interpreter.v3 [hunk 1]
```diff
@@ -2552,6 +2570,14 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 			genExtractLane(Opcode.F32X4_EXTRACTLANE, BpTypeCode.F32.code, byte.!(4), asm.movd_r_m, asm.movd_m_r, false);
 			genExtractLane(Opcode.F64X2_EXTRACTLANE, BpTypeCode.F64.code, byte.!(8), asm.movq_r_m, asm.movq_m_r, false);
 		}
+		{	// V128 lane replacement
+			genReplaceLane(Opcode.I8X16_REPLACELANE, byte.!(1), asm.movb_r_m, asm.movb_m_r);
```
Comment (titzer) [inline src/engine/x86-64/X86_64Interpreter.v3:2574]: The `byte.!` casts shouldn't be necessary, as the compiler will allow literals within range to be bytes.

**Review by titzer — COMMENTED** [review]
lgtm with minor cleanup

**Review by titzer — COMMENTED** [review]
lgtm with minor cleanup

**Comment by haoyu-zc** [comment]
Yes. I forgot that I changed the function's signature. Pushing a new commit.

**Review by titzer — APPROVED** [review]
lgtm

**Pull request merged**